service-proxy.js

```diff
@@ -950,7 +949,44 @@ function reapSinglePeer(hostPort) {
         self.deletePeerIndex(serviceName, hostPort);
     }
 
-    self.channel.peers.delete(hostPort);
+    // TODO: info log/stat
+
+    peer.drain({
+        reason: 'reaped for expired advertisement',
+        direction: 'both',
```

Comment:
wat is `onlyAccept: true` semantics?

Also prior was effectively `direction: 'both'` since it just did its own iteration of `peer.connections`... but sure if you want to change even more semantics in this diff, I'll do so ;-)